daffodil-runtime1/src/main/java/org/apache/daffodil/udf/UDFunctionService.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.udf;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.ServiceConfigurationError;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+
+import org.apache.daffodil.util.Misc;
+
+public class UDFunctionService {
```

Comment:
I've made the names to the spelled out versions. 

Indeed it is intended to load and cache UDFs at schema compile time. Yea, the reason it was implemented in Java was because I was having issues loading the classes from Scala. I tested it just now though, and I can access classes from Scala, so I wonder if my issues were due to a ServiceLoader learning curve. I'll update to the implementation to be in Scala and delete the java class. I'll also update with documentation.